src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1322,11 +1340,44 @@ theorem not_sshiftRight_not {x : BitVec w} {n : Nat} :
     ~~~((~~~x).sshiftRight n) = x.sshiftRight n := by
   simp [not_sshiftRight]
 
+theorem getMsbD_sshiftRight {x : BitVec w} {i n : Nat} :
+    getMsbD (x.sshiftRight n) i = (decide (i < w) && if i < n then x.msb else getMsbD x (i - n)) := by
+  simp only [getMsbD]
+  rw [BitVec.getLsbD_sshiftRight]
+  by_cases h : i < w
+  <;> by_cases h₁ : w ≤ w - 1 - i
+  <;> by_cases h₂ : ¬(i < n)
+  <;> by_cases h₃ : n + (w - 1 - i) < w
+  <;> by_cases h₄ : i - n < w
+  all_goals (simp [h, h₁, h₂, h₃, h₄]; try congr; try omega)
+  simp_all
```

Comment:
I am afraid this simp is non-terminal, so you may want to turn it into a simp only. You can try to be strategic and only include the minimal set of theorems needed to get it through.

Also, do you really need a simp_all at the end?